source/loader/layers/sanitizer/asan_options.hpp

```diff
@@ -0,0 +1,146 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file asan_options.hpp
+ *
+ */
+
+#pragma once
+
+#include "common/ur_util.hpp"
+#include "ur/ur.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+#include <algorithm>
+#include <cstring>
+#include <stdexcept>
+
+namespace ur_sanitizer_layer {
+
+struct AsanOptions {
+  public:
+    AsanOptions(AsanOptions &other) = delete;
+    void operator=(const AsanOptions &) = delete;
```

Comment:
```suggestion
    AsanOptions operator=(AsanOptions &) = delete;
```
Does copy constructor need to be deleted as well?

Reply:
oh, I missed the line above